components/brave_today/browser/brave_news_p3a_unittest.cc

```diff
@@ -233,20 +227,151 @@ TEST_F(BraveNewsP3ATest, TestTotalCardsViewed) {
 
   task_environment_.AdvanceClock(base::Days(4));
 
-  ResetCurrSessionTotalViewsCount(prefs);
+  RecordAtSessionStart(prefs);
   RecordTotalCardViews(prefs, 15);
   histogram_tester_.ExpectBucketCount(kTotalCardViewsHistogramName, 4, 1);
 
-  ResetCurrSessionTotalViewsCount(prefs);
+  RecordAtSessionStart(prefs);
   RecordTotalCardViews(prefs, 15);
   histogram_tester_.ExpectBucketCount(kTotalCardViewsHistogramName, 5, 1);
 
   task_environment_.AdvanceClock(base::Days(4));
 
-  ResetCurrSessionTotalViewsCount(prefs);
+  RecordAtSessionStart(prefs);
   RecordTotalCardViews(prefs, 0);
   histogram_tester_.ExpectBucketCount(kTotalCardViewsHistogramName, 4, 2);
 }
 
+TEST_F(BraveNewsP3ATest, TestLastUsageTime) {
+  PrefService* prefs = GetPrefs();
+  RecordAtInit(prefs);
+  // Should not report if News was never used
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 0);
+
+  RecordAtSessionStart(prefs);
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 1);
+  histogram_tester_.ExpectBucketCount(kLastUsageTimeHistogramName, 1, 1);
+
+  task_environment_.AdvanceClock(base::Days(7));
+  RecordAtInit(prefs);
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 2);
+  histogram_tester_.ExpectBucketCount(kLastUsageTimeHistogramName, 2, 1);
+
+  task_environment_.AdvanceClock(base::Days(7));
+  RecordAtInit(prefs);
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 3);
+  histogram_tester_.ExpectBucketCount(kLastUsageTimeHistogramName, 3, 1);
+
+  RecordAtSessionStart(prefs);
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 4);
+  histogram_tester_.ExpectBucketCount(kLastUsageTimeHistogramName, 1, 2);
+
+  task_environment_.AdvanceClock(base::Days(21));
+  RecordAtInit(prefs);
+  histogram_tester_.ExpectTotalCount(kLastUsageTimeHistogramName, 5);
+  histogram_tester_.ExpectBucketCount(kLastUsageTimeHistogramName, 4, 1);
+
+  task_environment_.AdvanceClock(base::Days(7));
+  RecordAtInit(prefs);
```

Comment:
What happens the the browser has been running continuously for 7 days. Would that be a `RecordAtSessionStart` without a previous `RecordAtInit`?

Reply:
If the browser is running continuously for multiple days, then `RecordAtInit` will only be called once. This means that metrics can become out of date if News hasn't been used in a while. I have an issue for this (brave/brave-browser#22441) and plan to make a PR shortly after this one is merged.